lib/interjectable.rb

```diff
@@ -42,6 +65,24 @@ def inject(dependency, &default_block)
           instance_variable_set(ivar_name, instance_eval(&default_block))
         end
       end
+
+      setter = :"#{dependency}="
+      if instance_variable_defined?(INJECTED_METHODS_IVAR)
+        instance_variable_set(INJECTED_METHODS_IVAR,
+                              (instance_variable_get(INJECTED_METHODS_IVAR) + [dependency, setter]).uniq)
+      else
+        instance_variable_set(INJECTED_METHODS_IVAR, [INJECTED_METHODS, dependency, setter])
+      end
+
+      injecting_class = self
+
+      define_method(INJECTED_METHODS) do
+        injecting_class.instance_variable_get(INJECTED_METHODS_IVAR)
+      end
+
+      define_singleton_method(INJECTED_METHODS) do
+        injecting_class.instance_variable_get(INJECTED_METHODS_IVAR)
+      end
```

Comment:
Yeah I think we should have these be methods that dynamically load instead of reading from a getter

I think we can also mirror the Ruby methods like [`Class#instance_methods`](https://docs.ruby-lang.org/en/master/Module.html#method-i-instance_methods) and have an optional `(include_super=true)`

Reply:
Sounds good! I like the idea of `(include_super=true)`, I'll look into adding it as well